http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/NonLeoMessageSubscriber.scala

```diff
@@ -213,7 +216,16 @@ class NonLeoMessageSubscriber[F[_]](config: NonLeoMessageSubscriberConfig,
   )(implicit ev: Ask[F, AppContext]): F[Unit] =
     for {
       ctx <- ev.ask
-      task = gkeAlg.deleteAndPollNodepool(DeleteNodepoolParams(msg.nodepoolId, msg.googleProject))
+      task = for {
+        _ <- gkeAlg.deleteAndPollNodepool(DeleteNodepoolParams(msg.nodepoolId, msg.googleProject))
+        apps <- appQuery.getAppsByNodepool(msg.nodepoolId).transaction
+        _ <- apps.traverse { app =>
+          authProvider
+            .notifyResourceDeleted(app.samResourceId, app.creator, msg.googleProject)
```

Comment:
It's okay to do this before the GKE deletion is finished, right?

Reply:
`gkeAlg.deleteAndPollNodepool` waits until the nodepool is deleted in GCP

Reply:
Oh, we don't have to do an `asyncTasks.offer()` call, the way we do below for the whole task?

Reply:
Ok, I think I get it - that async offer moves the whole thing to another thread, because it's moving through the steps sequentially. I wasn't sure if the `_ <-` flatMap would block further processing within the for-comp, or if there were some other Scala stuff happening.

Reply:
for comp always runs things 1. sequential 2. short-circuit in case of failure